src/crimson/osd/osd_operation.cc

```diff
@@ -149,27 +151,87 @@ void OSDOperationRegistry::visit_ops_in_flight(std::function<void(const ClientRe
   }
 }
 
+void OperationThrottler::start()
+{
+  LOG_PREFIX(OperationThrottler::start);
+  if (started) {
+    DEBUG("OperationThrottler background task is already started, skipping.");
+    return;
+  }
+
+  started=true;
+  stopped=false;
+
+  INFO("Starting OperationThrottler background task");
+  bg_future.emplace(background_task());
+  return;
+}
+
 OperationThrottler::OperationThrottler(ConfigProxy &conf)
-  : scheduler(crimson::osd::scheduler::make_scheduler(conf))
 {
   conf.add_observer(this);
-  update_from_config(conf);
 }
 
-void OperationThrottler::wake()
+void OperationThrottler::initialize_scheduler(CephContext *cct, ConfigProxy &conf, bool is_rotational, int whoami, MonClient *monc)
 {
-  while (available() && !scheduler->empty()) {
-    auto item = scheduler->dequeue();
-    item.wake.set_value();
-    ++in_progress;
-    --pending;
+  scheduler = crimson::osd::scheduler::make_scheduler(cct, conf, whoami, seastar::smp::count,
+            seastar::this_shard_id(), is_rotational, true, monc);
+  update_from_config(conf);
+}
+
+seastar::future<> OperationThrottler::background_task() {
```

Comment:
Commit message has nothing to do with this code seems like a leftover. Makes reviewing / and git blaming much more troublesome